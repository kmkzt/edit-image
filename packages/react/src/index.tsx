import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Counter } from '@/components/Counter'


export class App extends React.Component {
  constructor(props: any) {
    super(props)
  }

  public state = {
    count: 0
  }

  public increment = () => {
    this.setState({
      count: this.state.count + 1
    })
  }
  public decrement = () => {
    this.setState({
      count: this.state.count - 1
    })
  }

  public reset = () => {
    this.setState({
      count: 0
    })
  }

  public render() {
    const { count } = this.state
    return (
      <Counter
        count={count}
        counterIncrement={this.increment}
        counterDecrement={this.decrement}
        counterReset={this.reset}
      />
    )
  }
}

ReactDOM.render(
  <App />,
  document.getElementById('root') as HTMLElement
);
