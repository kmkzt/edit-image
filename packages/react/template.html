<!DOCTYPE html>
<html lang="ja">
  <head>
    <title>SAMPLE</title>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>
