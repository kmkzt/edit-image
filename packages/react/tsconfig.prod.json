{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "allowJs": false,
    "declaration": true,
    "declarationDir": "./types"
  },
  "exclude": ["./src/example"]
}
