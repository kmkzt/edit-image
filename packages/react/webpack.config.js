'use strict'

const { resolve } = require('path')
const { smart } = require('webpack-merge')
const devMode = process.env.NODE_ENV === 'development'
const config = devMode
  ? require('./webpack.dev.config')
  : require('./webpack.prod.config')

const common = {
  mode: devMode ? 'development' : 'production',

  module: {
    rules: [
      {
        test: /\.tsx?$/,
        exclude: /node_modules/,
        use: {
          loader: 'ts-loader',
          options: {
            configFile: devMode ? 'tsconfig.json' : 'tsconfig.prod.json',
            transpileOnly: devMode
          }
        }
      },
      {
        test: /\.s?css$/,
        exclude: /node_modules/,
        use: [
          'style-loader',
          'css-loader?modules&importLoaders=1&localIdentName=[path]___[name]__[local]___[hash:base64:5]',
          'resolve-url-loader',
          'sass-loader'
        ]
      },
      { test: /\.html$/, use: 'html-loader' }
    ]
  },

  resolve: {
    extensions: ['.ts', '.tsx', '.js', '.json'],
    alias: {
      '@': resolve(__dirname, 'src')
    }
  }
}

module.exports = smart(common, config)
