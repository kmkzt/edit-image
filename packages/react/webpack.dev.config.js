const { join, resolve } = require('path')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const Dotenv = require('dotenv-webpack')

const config = {
  entry: resolve('src', 'index.tsx'),
  devtool: 'source-map',

  output: {
    filename: '[name].bundle.js',
    path: resolve('public')
  },

  plugins: [
    new Dotenv({
      path: 'production.env',
      safe: false
    }),
    new HtmlWebpackPlugin({
      template: resolve('template.html')
    })
  ],

  devServer: {
    contentBase: join(__dirname, 'public'),
    compress: true,
    port: 9000
  }
}

module.exports = config
