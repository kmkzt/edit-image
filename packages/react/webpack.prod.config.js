const { join, resolve } = require('path')
const UglifyJsPlugin = require('uglifyjs-webpack-plugin')
const Dotenv = require('dotenv-webpack')

const config = {
  entry: resolve('src', 'library.tsx'),
  devtool: false,

  output: {
    filename: 'index.min.js',
    path: resolve('lib'),
    library: 'samplelibrary-reactts',
    libraryTarget: 'umd'
  },

  // 依存ライブラリの設定(使用先で必要なライブラリ)
  externals: {
    react: 'react',
    'styled-components': 'styled-components',
    'react-redux': 'react-redux',
    'react-router-dom': 'react-router-dom',
    'react-css-modules': 'react-css-modules'
  },
  plugins: [
    new Dotenv({
      path: 'production.env',
      safe: false
    })
  ],
  optimization: {
    minimizer: [
      new UglifyJsPlugin({
        parallel: true
      })
    ]
  }
}

module.exports = config
