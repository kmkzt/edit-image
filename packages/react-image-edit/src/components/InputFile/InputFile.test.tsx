import * as React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { InputFile } from '.'

describe('InputFile Test', () => {
  it('snapshot', () => {
    const tree = mount(<InputFile />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
