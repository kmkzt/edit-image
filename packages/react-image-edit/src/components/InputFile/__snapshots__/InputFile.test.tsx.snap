// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFile Test snapshot 1`] = `
<Component>
  <input
    accept="image/*"
    multiple={true}
    onChange={[Function]}
    type="file"
  />
</Component>
`;
